inlong-manager/manager-plugins/base/src/main/java/org/apache/inlong/manager/plugin/flink/FlinkOperation.java

```diff
@@ -351,20 +350,21 @@ public void pollJobStatus(FlinkInfo flinkInfo) throws Exception {
                 }
 
                 JobStatus jobStatus = jobDetailsInfo.getJobStatus();
-                if (jobStatus.isTerminalState()) {
+                if (jobStatus.isTerminalState() && expectStatus != JobStatus.CANCELED) {
                     log.error("job was terminated for {}, exception: {}", jobId, flinkInfo.getExceptionMsg());
                     throw new Exception("job was terminated for " + jobId);
                 }
 
-                if (jobStatus == RUNNING) {
-                    log.info("job status is Running for {}", jobId);
+                if (jobStatus == expectStatus) {
+                    log.info("job status is {} for {}", jobStatus, jobId);
                     break;
                 }
-                log.info("job was not Running for {}", jobId);
-                TimeUnit.SECONDS.sleep(5);
+                log.info("job status is {} for {}", jobStatus, jobId);
             } catch (Exception e) {
                 log.error("poll job status error for {}, exception: ", flinkInfo, e);
             }
+            TimeUnit.SECONDS.sleep(5);
```

Comment:
why sleep 5 ms

Reply:
Because it is necessary to continuously loop to determine whether the job status is expected.